spring-vault-core/src/main/java/org/springframework/vault/authentication/LdapAuthentication.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.vault.authentication;
+
+import org.springframework.web.client.RestOperations;
+
+/**
+ * LDAP implementation of {@link ClientAuthentication}.
+ *
+ * @author Mikhael Sokolov
+ * @see LdapAuthenticationOptions
+ * @see RestOperations
+ * @see <a href="https://www.vaultproject.io/docs/auth/ldap">LDAP</a>
+ * @since 2.4
+ */
+public class LdapAuthentication extends UsernamePasswordAuthentication {
```

Comment:
Let's not subclass `UsernamePasswordAuthentication`. While username/password are the primary driver for LDAP, future implementations could bring alternative schemes for LDAP and we should not constrain ourselves.

Reply:
I agree, fixed